parser/src/main/java/com/linkedin/avroutil1/parser/avsc/AvscParser.java

```diff
@@ -246,19 +286,20 @@ private SchemaOrRef parseComplexSchema(
                     + typeStr.getLocation() + ". expecting \"record\", \"enum\" or \"fixed\"");
         }
 
+        LinkedHashMap<String, JsonValueExt> propsMap = parseExtraProps(objectNode, CORE_SCHEMA_PROPERTIES);
+        JsonPropertiesContainerImpl props = new JsonPropertiesContainerImpl(propsMap);
```

Comment:
Why not do the same thing as you do on line 401? I.e. if the props is empty, use JsonPropertiesContainer.EMPTY instead of creating an object of the impl.